packages/dds/map/src/directory.ts

```diff
@@ -811,6 +752,79 @@ export class SharedDirectory extends SharedObject<ISharedDirectoryEvents> implem
     protected applyStashedOp() {
         throw new Error("not implemented");
     }
+
+    private serializeDirectory(
+        root: SubDirectory,
+        serializer: IFluidSerializer,
+        telemetryContext?: ITelemetryContext,
+    ): ISummaryTreeWithStats {
+        const MinValueSizeSeparateSnapshotBlob = 8 * 1024;
+
+        const builder = new SummaryTreeBuilder();
+        let counter = 0;
+        const blobs: string[] = [];
+
+        const stack: [SubDirectory, IDirectoryDataObject][] = [];
+        const content: IDirectoryDataObject = {};
+        stack.push([root, content]);
+
+        this.incrementSummarizeInstanceCount(telemetryContext);
+
+        let totalBlobBytes = 0;
+
+        while (stack.length > 0) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const [currentSubDir, currentSubDirObject] = stack.pop()!;
+            for (const [key, value] of currentSubDir.getSerializedStorage(serializer)) {
+                if (!currentSubDirObject.storage) {
+                    currentSubDirObject.storage = {};
+                }
+                const result: ISerializableValue = {
+                    type: value.type,
+                    // eslint-disable-next-line @typescript-eslint/ban-types
+                    value: value.value && JSON.parse(value.value) as object,
+                };
+                if (value.value && value.value.length >= MinValueSizeSeparateSnapshotBlob) {
+                    const extraContent: IDirectoryDataObject = {};
+                    let largeContent = extraContent;
+                    if (currentSubDir.absolutePath !== posix.sep) {
+                        for (const dir of currentSubDir.absolutePath.substr(1).split(posix.sep)) {
+                            const subDataObject: IDirectoryDataObject = {};
+                            largeContent.subdirectories = { [dir]: subDataObject };
+                            largeContent = subDataObject;
+                        }
+                    }
+                    largeContent.storage = { [key]: result };
+                    const blobName = `blob${counter}`;
+                    counter++;
+                    blobs.push(blobName);
+                    builder.addBlob(blobName, JSON.stringify(extraContent));
+                    totalBlobBytes += value.value.length;
```

Comment:
>totalBlobBytes [](http://example.com/codeflow?start=20&length=14)

same as before - If I'm reading it right we would need to increment totalBlobBytes on the else as well. W will call addBLob on 822 again ?